symphony-bdk-legacy/sms-sdk-renderer-java/src/test/java/it/services/SmsRendererTest.java

```diff
@@ -0,0 +1,160 @@
+
+    String jsonMessageContext = null;
+    JSONArray jsonArray = null;
+
+    if (smsTypes != SmsRenderer.SmsTypes.TABLE) {
+      final String fileName = typeName + "JsonArray.json";
+      jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+      jsonArray = new JSONArray();
+      jsonArray.add(jsonMessageContext);
+    } else {
+      final JSONParser parser = new JSONParser();
+      final String sourceFile = System.getProperty("user.dir") + "/src/test/resources/SmsRenderer/JSONData/tableJsonArray.json";
+      final Object obj = parser.parse(new FileReader(sourceFile));
+      final JSONObject jsonObject = (JSONObject) obj;
+      jsonArray = (JSONArray) jsonObject.get("phones");
+    }
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonArray, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "JsonArrayHtml.html");
+  }
+
+  private void verifyResultJSON(final String typeName, final String compiledTemplate, final String fileName) throws IOException {
+    final String expectedResult = this.minifyHTML(this.getStringFromFile("/SmsRenderer/HTMLResult/" + typeName + fileName));
+    assertNotNull(expectedResult);
+    assertEquals(expectedResult, compiledTemplate);
+  }
+
+  private void testString(final SmsRenderer.SmsTypes smsTypes) throws IOException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String jsonFileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + jsonFileName);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonMessageContext, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "StringHtml.html");
+  }
+
+  private void testJsonObject(final SmsRenderer.SmsTypes smsTypes) throws IOException, ParseException {
+    assertNotNull(smsTypes);
+
+    final String typeName = smsTypes.getName().toLowerCase();
+    final String fileName = typeName + "Json.json";
+    final String jsonMessageContext = this.getStringFromFile("/SmsRenderer/JSONData/" + fileName);
+
+    final JSONObject jsonObject = this.getJsonObject(jsonMessageContext);
+    assertNotNull(jsonObject);
+
+    final String compiledTemplate = this.minifyHTML(SmsRenderer.renderInBot(jsonObject, smsTypes));
+    assertNotNull(compiledTemplate);
+
+    this.verifyResultJSON(typeName, compiledTemplate, "JsonObjectHtml.html");
+  }
+
+  private String getStringFromFile(final String fileName) throws IOException {
+    assertNotNull(fileName);
+
+    final String jsonMessageContext = this.removeCarriageReturn(this.readResourceContent(fileName));
+    assertNotNull(jsonMessageContext);
+    return jsonMessageContext;
+  }
+
+  private String readResourceContent(final String path) throws IOException {
+    assertNotNull(path);
```

Comment:
If you want to keep this way of reading files, please remove the `assertNotNull` statements, they don't add any value because there are not directly related the SmsRenderer.